src/Models/NonhydrostaticModels/velocity_and_tracer_tendencies.jl

```diff
@@ -5,15 +5,27 @@ using Oceananigans.Operators
 using Oceananigans.StokesDrift
 using Oceananigans.TurbulenceClosures: ∂ⱼ_τ₁ⱼ, ∂ⱼ_τ₂ⱼ, ∂ⱼ_τ₃ⱼ, ∇_dot_qᶜ
 
+"return the ``x``-gradient of hydrostatic pressure"
+hydrostatic_pressure_gradient_x(i, j, k, grid, hydrostatic_pressure) = ∂xᶠᶜᵃ(i, j, k, grid, hydrostatic_pressure)
+hydrostatic_pressure_gradient_x(i, j, k, grid, ::Nothing) = 0
```

Comment:
```suggestion
hydrostatic_pressure_gradient_x(i, j, k, grid, ::Nothing) = zero(eltype(grid))
```

might be a bit safer